src/proxy/momento/src/frontend.rs

```diff
@@ -75,6 +78,7 @@ pub(crate) async fn handle_memcache_client_concrete(
         + Send
         + 'static,
     flags: bool,
+    proxy_metrics: Arc<impl ProxyMetricsApi>,
```

Comment:
nit: given that you’re using a trait, you could just accept impl ProxyMetrics here.

you could then make `ProxyMetrics: Clone + Send + Sync + ‘static` and impl ProxyMetrics for `Arc<Whatever>`. Would clean up the parameter lists anyway.

Reply:
On nice. Changed in [0d44c63](https://github.com/pelikan-io/pelikan/pull/149/commits/0d44c63436e390b236af4d879befd4294c3eae87). Very clean!